src/coffee/import.coffee

```diff
@@ -498,7 +503,15 @@ class Import
       else
         Promise.resolve "[row #{rowIndex}] Product update not necessary."
 
+  assignState: (product) ->
+    if (@states.key2id[@defaultState])
+      product.state = { typeId : "state", id : @states.key2id[@defaultState] }
```

Comment:
When creating categories, it is sufficient to provide `key` instead of id.. so you don't have to load unnecessarily resources from API.. Maybe it works also here with states:
```js
+      product.state = { typeId : "state", key: @defaultState }
```

Reply:
So it does... though the fetching of state by the Validator was already there in the code, introduced in May by Williams, and the fetching of Validator resources in general as part of `import` was introduced by you a couple of years ago.

Reply:
...and without that, it becomes impossible to tell in advance if the default state actually exists, and therefore, if not, not add it to the product draft so as not to cause product creation to fail.

Reply:
Also, the reason "Import integration test"."should set default state on update when configured to do so, and product lacks state" is failing turns out to be because sync-actions relies on `id` being present, so I will revert this change.